homeassistant/components/http/ban.py

```diff
@@ -110,6 +110,12 @@ def log_invalid_auth(func):
 
     request.app[KEY_FAILED_LOGIN_ATTEMPTS][remote_addr] += 1
 
+    # Supervisor IP should never be banned
+    if "hassio" in hass.config.components and hass.components.hassio.get_supervisor_ip() == str(
```

Comment:
Another solution I thought of now, is to allow integrations to register a ban exclusion callback to the http ban. That would avoid adding other integrations code into the ban module here.

I think it's apparent when looking at the tests that it's weird that we need to set up the hassio integration to test the code in the ban module. It would be better if we added the test to the hassio integration and tested that the supervisor ip is never banned.

Reply:
Well, Hass.io is Home Assistant - so it should be fine if we make sure that we don't shoot us itself

Reply:
This is a general concept, that we should follow regardless of the status of the integration. It's just a matter of dependency flow. It's the hassio integration that should be responsible to set up the ip ban exclusion for its ip address. The ban module is just the facilitator of that exclusion.